### PR TITLE
Bug 1906808: follow-up fixes to SA-issuer observer

### PR DESCRIPTION
**Pull request by stlaz**
This PR modifies the SA-issuer observer to match how the other observers are behaving to simplify maintenance.

It's a follow-up to https://github.com/openshift/cluster-kube-apiserver-operator/pull/1020, per request in https://github.com/openshift/cluster-kube-apiserver-operator/pull/1020#discussion_r551799962, I like to believe the code flow is easier to follow now, even for me as I was getting lost in the previous form of it.

/assign @sttts

### CODE REVIEW

File: pkg/operator/configobservation/auth/auth_serviceaccountissuer.go [hunk 1]
```diff
@@ -5,133 +5,128 @@ import (
...
+	if err != nil {
+		errs = append(errs, fmt.Errorf("unable to extract service account issuer from unstructured: %v", err))
+	}
 
-	previousIssuer, moreErrs := issuerFromUnstructuredConfig(existingConfig)
-	errs = append(errs, moreErrs...)
+	if existingIssuers != nil && len(existingIssuers) > 0 {
+		existingIssuer = existingIssuers[0]
+	} else {
+		existingIssuer = ""
```
Comment (sttts) [inline pkg/operator/configobservation/auth/auth_serviceaccountissuer.go:69]: this is empty anyway

**Comment by stlaz** [comment]
/bugzilla refresh
rebased on master w/o any changes

**Pull request merged**